check-pull-request-status.py

```diff
@@ -23,14 +24,34 @@
 pull_request = client.pull_request(owner='openshift', repository='openshift-ansible', number=pull_request_id)
 pull_request_statuses = client._json(client._get(pull_request.statuses_url.replace('statuses', 'status')), 200)['statuses']
 
+# This list has tu be updated for a new version after each release cut
+versions = ["3.6", "3.5", "3.4", "3.3", "3.2", "3.1"]
+
+prefix = "aos-ci-jenkins/OS_"
+suffix_variations = ["_NOT_containerized",
+    "_NOT_containerized_e2e_tests",
+    "_containerized",
+    "_containerized_e2e_tests"
+]
+
 blocking_statuses = [
-    'aos-ci-jenkins/OS_3.5_NOT_containerized',
-    'aos-ci-jenkins/OS_3.5_NOT_containerized_e2e_tests',
-    'aos-ci-jenkins/OS_3.5_containerized',
-    'aos-ci-jenkins/OS_3.5_containerized_e2e_tests',
     'continuous-integration/travis-ci/pr'
 ]
 
+# Master tests current devel (3.6) and last release (3.5)
+if target_branch == "master":
+    for version in versions[:2]:
+        for suffix in suffix_variations:
+            blocking_statuses.append(prefix + version + suffix)
+# Other branches eg: release-1.5 only test their respective versions
+elif target_branch.startswith("release-"):
```

Comment:
This presumes that the statuses exist at the same moment that we break out a new branch and run a `[test]` against it. I don't think that generally this is true. I think a better approach would be to accept a external config that would have a static mapping of branch to tests. This would allow for us to be certain that we are not changing what tests we look for unless someone explicitly adds or removes one from the config.